### PR TITLE
Add API to retrieve installation info

### PR DESCRIPTION
**Pull request by niboshi**
Adding API to retrieve installation info, such as include/library directory paths.

Needed for https://github.com/chainer/chainer/pull/7442

### CODE REVIEW

File: cupyx/runtime.py [hunk 1]
```diff
@@ -23,6 +25,43 @@ def _eval_or_error(func, errors):
         return repr(e)
 
 
+class _InstallInfo(object):
+
+    def __init__(self):
+        cupy_package_root = self._get_cupy_package_root()
```
Comment (emcastillo) [inline cupyx/runtime.py]: Just a suggestion, I think we can use pip to find the package_root  and wether the package was installed using pip itself.

from pip._internal.utils.misc import get_installed_distributions
filter(lambda x: 'cupy' in x.project_name, get_installed_distributions())

The returned object has a 'location' attribute to get the install path

**Comment by emcastillo** [comment]
@niboshi as I understand you are not planning to address the is_binary issue in this PR?

**Comment by niboshi** [comment]
@emcastillo No, not in this PR (I don't immediately come up with the correct way to provide it.)

**Comment by niboshi** [comment]
@emcastillo 
Oops, please do not forget to trigger CIs!

**Pull request merged**